### PR TITLE
Enforce specification of subjects_dir if deriv_root is set

### LINKED ISSUE #404: FreeSurfer step doesn't respect deriv_root
Instead, it will write outputs to `bids_root/derivatives/freesurfer` no matter what.

Reported by @dengemann

### PR DESCRIPTION
**Pull request by hoechenberger**
Didn't work properly before: recon-all output was always written to `SUBJECTS_DIR/`.

Fixes #404

### Before merging …

- [ ] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

**Comment by agramfort** [comment]
Not sure about this. In ds117 they ship a free surfer recon but it’s incomplete. Rather than deleted their files I’ve changed the subjects_dir. With this fix I would be stuck

**Comment by hoechenberger** [comment]
I'm not sure I fully understand. Maybe let's have a call tonight if you have time (I'm currently out to have coffee :))

**Comment by hoechenberger** [comment]
Btw motivation was that @dengemann had sets deriv_root that deviated from the default (bids_root/derivatives), yet our FS script would produce the  recon-all output in bids_root/derivatives/freesurfer, which was clearly the wrong place

**Comment by hoechenberger** [comment]
@agramfort I think I got what you meant and have started a new attempt. WDYT?

File: config.py [hunk 1]
```diff
@@ -59,13 +59,26 @@
...
-derivativesfor all subjects. Specifically, the ``subjects_dir`` is the
-$SUBJECTS_DIR used by the Freesurfer software.
+Path to the directory that contains the FreeSurfer parcellations of all
+subjects. Specifically, this defines the ``SUBJECTS_DIR`` that is used by
+FreeSurfer. When running the ``freesurfer`` processing step to create the
+parcellations from anatomical scans in the BIDS dataset, the output will be
+stored in this directory.
+
+If ``None``, this will default
+to [`bids_root`][config.bids_root_root]`/freesurfer/subjects`.
```
Comment (hoechenberger) [inline config.py:77]:
```suggestion
to [`bids_root`][config.bids_root_root]`/derivatives/freesurfer/subjects`.
```

**Comment by hoechenberger** [comment]
Ok should be good now.

File: config.py [hunk 2]
```diff
@@ -59,13 +59,26 @@
...
+    If specified and you wish to run the source analysis steps, you must
+    set [`subjects_dir`][config.subjects_dir] as well.
 """
 
 subjects_dir: Optional[PathLike] = None
 """
-Path to the directory that contains the MRI data files and their
-derivativesfor all subjects. Specifically, the ``subjects_dir`` is the
-$SUBJECTS_DIR used by the Freesurfer software.
+Path to the directory that contains the FreeSurfer parcellations of all
```
Comment (agramfort) [inline config.py:70]: It’s the « reconstructions » not the parcellations. To fix here and below.

**Review by agramfort — COMMENTED** [review]
Besides LGTM !

File: config.py (hunk 2 above)
Comment (hoechenberger) [reply config.py:70]: Right, thanks!

**Pull request merged**